prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt

```diff
@@ -328,29 +316,62 @@ class FHIRRouter(
             // JURIS FILTER
             //  default: allowNone
             var passes = evaluateFilterCondition(getJurisFilters(receiver, orgFilters), bundle, false)
+
             // QUALITY FILTER
-            //  default: must have message id, patient last name, patient first name, dob, specimen type
-            //           must have at least one of patient street, zip code, phone number, email
-            //           must have at least one of order test date, specimen collection date/time, test result date
-            passes = passes &&
-                evaluateFilterCondition(
-                    getQualityFilters(receiver, orgFilters),
+            if (passes) {
+                //  default: must have message id, patient last name, patient first name, dob, specimen type
+                //           must have at least one of patient street, zip code, phone number, email
+                //           must have at least one of order test date, specimen collection date/time, test result date
+                // get the quality filter default result for the bundle, but only if it is needed
+                val qualFilterDefaultResult: Boolean by lazy {
```

Comment:
Why did you move this? The default value is one-per-bundle, regardless of which receiver it is for, and should not be taking 'receiver. resverTheQualityFilter into account since this would result in a double negation - it would negate the default and then negate that negation when it ran the evaluation.

Reply:
I had done that to handle the logging before understanding it better. I've reverted those changes.